packages/host/app/components/card-catalog/modal.gts

```diff
@@ -213,44 +216,50 @@ export default class CardCatalogModal extends Component<Signature> {
     // returns all available realms and their cards that match a certain type criteria
     // realm filters and search key filter these groups of cards
     // filters dropdown menu will always display all available realms
-    if (this.currentRequest?.search.instancesByRealm.length) {
-      this.searchResults = this.currentRequest?.search.instancesByRealm;
+    if (this.request?.search.instancesByRealm.length) {
+      this.state.searchResults = this.request?.search.instancesByRealm;
     }
-    return this.currentRequest?.search.instancesByRealm ?? [];
+    return this.request?.search.instancesByRealm ?? [];
   }
 
   get displayedRealms(): RealmCards[] {
     // filters the available realm cards by selected realms
-    return this.selectedRealms.length
-      ? this.selectedRealms
+    return this.state.selectedRealms.length
+      ? this.state.selectedRealms
       : this.availableRealms;
   }
 
-  _selectedRealms = new TrackedArray<RealmCards>([]);
+  get request(): Request | undefined {
+    return this.state.request;
+  }
+
+  get state(): State {
+    if (this.stateStack.length <= 0) {
+      return {} as State;
```

Comment:
how is this a valid state? in your `State` type there are fields thgat are required. perhaps it would be better to return undefined than an empty object--cause an empty object violates oiur `State` type (and when we do that we lose all the type guarantees that TS gives us). 

Returning undefined here would answer my questions around why you check for this.state to exist